Documentation/artifact-format.md

```diff
@@ -175,8 +273,89 @@ version is specified outside of `header.tar.gz`.
 
 Why is there a tar file inside a tar file? See the "Ordering" section.
 
+### header-info (version 3)
+
+Format: JSON
+
+`header-info` must be the first file within `header.tar.gz`. Its content is:
+
+```
+{
+  "updates": [
+    {
+      "type": "rootfs-image"
+    },
+    {
+      "type": "rootfs-image"
+    },
+    {
+      ...
+    }
+  ],
+  "artifact_provides": [
+          { "artifact_name": "name" },
+          { "artifact_group": "group-1" },
+          { "update_types_supported": ["rootfs-image"]},
+          { "rootfs_image_checksum": "1d0b820130ae028ce8a79b7e217fe505a765ac394718e795d454941487c53d32" },
+          { "provides_type": ["provides_value", "provides_value"] },
+      ],
+  "artifact_depends": [
+          { "device_type": ["vexpress-qemu", "beaglebone"] },
+          { "artifact_name": ["rootfs-1"] },
+          { "rootfs_image_checksum": "4d480539cdb23a4aee6330ff80673a5af92b7793eb1c57c4694532f96383b619" },
+      ],
+  ],
+}
+```
+
+The `updates` parameter is as described in the
+section below (see the version 2 header-info description for the details).
+
+#### artifact_depends
+
+The `artifact_depends` contains a set of parameters that the current artifact
+depends on. It can contain zero or more elements (in most cases at least
+`device_type` should be present though).
+
+The given artifact will be installed, only if the device itself
+and the artifact currently installed on the device are providing a full set
+of matching parameters. The complete list contains following parameters:
+
+* `artifact_name` is the name of the artifact currently installed on the device
+* `rootfs_image_checksum` is the checksum of the image that needs to be installed
+on the device before current artifact can be installed
+* `device_type` is the type of the device (see `device_provides` below)
+* `artifact_group` is the group of the artifacts current artifact belongs to
+* `update_types_supported` is the list of the update types Mender agent installed
+on the device can handle
+* `artifact_versions_supported` is the list of the versions of the artifacts
+agent installed on the device can handle
```

Comment:
`artifact_versions_supported` is not in your list above, and I think it is redundant, because the `version` header already has this number in it. It can be present in the Provides section, however.

Reply:
But don't we need this for depends? For provides we will use the `version`, but we need to have some kind of information if the given artifact can be installed (if depends only on certain version).